lib/permutations.rb

```diff
@@ -1,4 +1,21 @@
 
 def permutations?(string1, string2)
-  
+  l1 = string1.length
+  l2 = string2.length
+  return false if l1 != l2
+
+  hash = {}
+  i = 0
+  while i < l1
+    hash[string1[i]] = true
+    i += 1
```

Comment:
As with the other two methods, try to think of a variable name that is more descriptive of how the hash is used and what is stored in it.